scrapy/spiders/crawl.py

```diff
@@ -65,9 +67,19 @@ class CrawlSpider(Spider):
     def __init__(self, *a, **kw):
         super().__init__(*a, **kw)
         self._compile_rules()
+        cls = self.__class__
+        self._is_warned = False
+        is_overriden = method_is_overridden(cls, CrawlSpider, "_parse_response")
+        if is_overriden:
+            warnings.warn(
+                "The CrawlSpider._parse_response method is deprecated: it "
+                "will be removed in future Scrapy releases. Please use the "
+                "CrawlSpider.parse_with_rules method instead."
+            )
```

Comment:
What about mentioning where the problem is? Users may not realize they are subclassing CrawlSpider, or where, or they may be using a Scrapy plugin, or subclassing a CrawlSpider subclass from a third-party, etc.

```suggestion
            cls_path = f"{cls.__module__}.{cls.__qualname__}"
            warnings.warn(
                f"The CrawlSpider._parse_response method, that the "
                f"{cls_path} class reimplements, is deprecated: it will be "
                f"removed in future Scrapy releases. Please use the "
                f"CrawlSpider.parse_with_rules method instead."
            )
```